File: 백준/Silver/1406. 에디터/에디터.java
import java.io.*;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        String input = br.readLine();
        int N = Integer.parseInt(br.readLine());
        Stack<String> leftStack = new Stack<>();
        Stack<String> rightStack = new Stack<>();
        for (int i = 0; i < input.length(); i++) {
            leftStack.push(String.valueOf(input.charAt(i)));
        }
        for (int i = 0; i < N; i++) {
            String command = br.readLine();
            if (command.contains("L")) {
                if (!leftStack.isEmpty()) {
                    String item = leftStack.pop();
                    rightStack.push(item);
                }
            } else if (command.contains("D")) {
                if (!rightStack.isEmpty()) {
                    String item = rightStack.pop();
                    leftStack.push(item);
                }
            } else if (command.contains("B")) {
                if (!leftStack.isEmpty()) {
                    leftStack.pop();
                }
            } else if (command.contains("P")) {
                String item = command.substring(2);
                leftStack.push(item);
            }
        }
        int size = rightStack.size();
        for (int i = 0; i < size; i++) {
            leftStack.push(rightStack.pop());
        }
        StringBuilder sb = new StringBuilder();
        size = leftStack.size();
        for (int i = 0; i < size; i++) {
            sb.append(leftStack.pop());
        }
        bw.write(String.valueOf(sb.reverse()));
        bw.flush();
    }
}

File: 백준/Silver/1406. 에디터/에디터.py
from sys import stdin as s

# input.txt file
#s = open('input.txt')

input = s.readline().rstrip()
n = int(s.readline().rstrip())

rightStack = [0] * len(input)
leftStack = []
for i in range(0, len(input)):
    rightStack[i] = input[i]

#입력 처리
for i in range(0, n):
    command = s.readline().rstrip()
    if command[0] == "L":
        if rightStack:
            leftStack.append(rightStack.pop())
    elif command[0] == "D":
        if leftStack:
            rightStack.append(leftStack.pop())
    elif command[0] == "B":
        if rightStack:
            rightStack.pop()
    elif command[0] == "P":
        rightStack.append(command[2])

leftStack.reverse()
resultList = rightStack + leftStack
print(''.join(resultList))
